persistence/src/main/scala/akka/persistence/EventStoreinterop.scala

```diff
@@ -0,0 +1,196 @@
+    * before they will be consumed (ie deleted from buffer) as [[EventStore.events]] stream. The output stream is lazy by itself and actual
+    * event consumption from the buffer will happened only on the stream materialization.
+    *
+    * @param system
+    *   Akka system
+    * @param timeout
+    *   maximum time between messages from Akka' journal plugin (is the next message expected)
+    * @param capacity
+    *   internal event buffer capacity, on oveflow will raise [[BufferOverflowException]]
+    * @param journalPluginId
+    *   Akka persistence journal plugin ID
+    * @param eventSourcedId
+    *   Akka persistence ID, unique per each actor
+    * @return
+    *   instance of [[EventStore]]
+    */
+  def apply[F[_]: Async: FromFuture: ToTry, A](
+    system: ActorSystem,
+    timeout: FiniteDuration,
+    capacity: Int,
+    journalPluginId: String,
+    eventSourcedId: EventSourcedId
+  ): F[EventStore[F, A]] =
+    Sync[F]
+      .delay {
+        val actorRef = Persistence(system).journalFor(journalPluginId)
+        ActorEffect.fromActor(actorRef)
+      }
+      .map { journaller =>
+        new EventStore[F, A] {
+
+          import sstream.FoldWhile._
+
+          val persistenceId = eventSourcedId.value
+
+          override def events(fromSeqNr: SeqNr): F[sstream.Stream[F, EventStore.Persisted[A]]] = {
+
+            type Buffer = Vector[EventStore.Event[A]]
+
+            def actor(buffer: Ref[F, Buffer]) =
+              LocalActorRef[F, Unit, SeqNr]({}, timeout) {
+
+                case (_, JournalProtocol.ReplayedMessage(persisted)) =>
+                  if (persisted.deleted) ().asLeft[SeqNr].pure[F]
+                  else {
+                    val payload = persisted.payload.asInstanceOf[A]
+                    val event   = EventStore.Event(payload, persisted.sequenceNr)
+
+                    buffer
+                      .modify { buffer =>
+                        val buffer1 = buffer :+ event
+                        val lenght  = buffer.length + 1
+                        buffer1 -> lenght
+                      }
+                      .flatMap { lenght =>
+                        if (lenght > capacity) {
+                          new BufferOverflowException(capacity, persistenceId).raiseError[F, Either[Unit, SeqNr]]
+                        } else {
+                          ().asLeft[SeqNr].pure[F]
+                        }
+                      }
+                  }
+
+                case (_, JournalProtocol.RecoverySuccess(seqNr)) =>
+                  seqNr.asRight[Unit].pure[F]
+
+                case (_, JournalProtocol.ReplayMessagesFailure(error)) =>
+                  error.raiseError[F, Either[Unit, SeqNr]]
+              }
+
+            for {
+              buffer <- Ref[F].of[Buffer](Vector.empty)
+              actor  <- actor(buffer)
+              request = JournalProtocol.ReplayMessages(fromSeqNr, SeqNr.Max, Long.MaxValue, persistenceId, actor.ref)
```

Comment:
I would consider using named parameters here, when constructing `ReplayMessage`, as it will be too easy to mix up first three `Long` parameters when doing refactoring.